src/data.rs

```diff
@@ -450,6 +453,99 @@ pub fn decode_block_cid_ask_message(msg: Vec<u8>) -> Option<(i128, Option<Cid>)>
 	}
 }
 
+pub async fn ipfs_priority_get_cells(
+	cells: &mut Vec<DataCell>,
+	ipfs: &Ipfs<DefaultParams>,
+	rpc_url: &String,
+	block_num: u64,
+) -> Result<()> {
+	println!("ipfs priority get started...");
+	// Try fetching the cells from IPFS first
+	for (_, req_cell_data) in cells.iter_mut().enumerate() {
+		let peers = ipfs.peers();
+		println!("number of peers: {}", peers.len());
+		if peers.len() == 0 {
+			break;
+		}
+		// println!("list of connections: {:?}", ipfs.connections());
+		// TODO: paralelize fetching
+		// Skip cells that return error when fetching from IPFS
+		println!(
+			"Requesting from IPFS: CID: {}. Cell reference: {}.",
+			req_cell_data.cid(),
```

Comment:
This is interesting :) Since this struct is used here to create request, cid is not valid (data =[]).